contracts/optimism/TokenRateOracle.sol

```diff
@@ -116,21 +134,19 @@ contract TokenRateOracle is CrossDomainEnabled, ITokenRateOracle, Versioned {
             revert ErrorL1TimestampExceededAllowedClockLag(tokenRate_, rateL1Timestamp_);
         }
 
-        /// @dev use only the more actual token rate
+        /// @dev Use only the most up-to-date token rate. Reverting should be avoided as it may occur occasionally.
         if (rateL1Timestamp_ <= _getRateL1Timestamp()) {
             emit DormantTokenRateUpdateIgnored(tokenRate_, rateL1Timestamp_, _getRateL1Timestamp());
             return;
         }
 
         /// @dev allow token rate to be within some configurable range that depens on time it wasn't updated.
-        if (!_isTokenRateWithinAllowedRange(tokenRate_, rateL1Timestamp_)) {
+        if (tokenRate_ != _getTokenRate() && !_isTokenRateWithinAllowedRange(tokenRate_, rateL1Timestamp_)) {
```

Comment:
why not retrieve `_getTokenRate()` once inside the `_isTokenRateWithinAllowedRange` method? This can be easily optimized, please don't read from the same storage multiple times, use local vars instead